src/config/config_poll.rs

```diff
@@ -0,0 +1,84 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Duration;
+
+use url::Url;
+
+use crate::blueprint::Blueprint;
+use crate::config::reader::ConfigReader;
+use crate::config::Config;
+use crate::http::ServerContext;
+
+pub struct ConfigLoader {
+  file_paths: HashMap<String, u64>,
+  poll_interval: u64,
+}
+
+impl ConfigLoader {
+  pub fn init(file_paths: Vec<String>, poll_interval: u64) -> ConfigLoader {
+    let mut path_hm = HashMap::new();
+    for file_path in file_paths {
+      path_hm.insert(file_path, poll_interval);
+    }
+    Self { file_paths: path_hm, poll_interval }
+  }
+  pub async fn start_polling(&mut self, batch_single: Arc<ServerContext>) {
+    loop {
```

Comment:
Instead of `loop` it would be better to only query every `n` units of time. Internally keep a last fetch time and request it only if the time has expired. Essentially behaving as TTL and not Polling. This way we don't need to worry about leaks and using `drop`.

Reply:
The other advantage with using a TTL is the ability to set that in the config instead of passing it via the CLI.

Reply:
> Instead of `loop` it would be better to only query every `n` units of time. Internally keep a last fetch time and request it only if the time has expired. Essentially behaving as TTL and not Polling. This way we don't need to worry about leaks and using `drop`.

So you mean that TTL can be checked when user queries something and if it's expired then we try to fetch new schema?